jupyterhub_chameleon/handler.py

```diff
@@ -213,86 +216,40 @@ async def get(self):
         await self.refresh_token(source="artifact_publish_prepare")
 
         auth_state = await self.current_user.get_auth_state()
-        openstack_rc = auth_state.get(OPENSTACK_RC_AUTH_STATE_KEY, {})
-        user_session = keystone_session(env_overrides=openstack_rc)
-
-        admin_overrides = {
-            key.replace("ARTIFACT_SHARING_", ""): value
-            for key, value in os.environ.items()
-            if key.startswith("ARTIFACT_SHARING_OS_")
-        }
-        admin_session = keystone_session(env_overrides=admin_overrides)
-        # NOTE(jason): we have to set interface/region_name explicitly because
-        # Keystone does not read these from the session/adapter.
-        admin_ks_client = KeystoneClient(
-            session=admin_session,
-            interface=os.environ.get("OS_INTERFACE"),
-            region_name=os.environ.get("OS_REGION_NAME"),
-        )
 
-        response = None
-        trust_project_name = os.environ.get(
-            "ARTIFACT_SHARING_TRUST_PROJECT_NAME", "trovi"
+        # Authenticate to Trovi with our OpenStack credentials,
+        # exchanging our keystone token for a Trovi token
+        trovi_resp = requests.post(
+            urljoin(os.getenv("TROVI_URL"), "/token/"),
+            headers={"Content-Type": "application/json", "Accept": "application/json"},
```

Comment:
fun fact: headers are case-insensitive, it's not necessary to capitalize them. Just a note for the future in case it comes up, took me a long time to learn/realize this!